code_challenges/code_wars/highest_and_lowest.py

```diff
@@ -0,0 +1,16 @@
+# https://www.codewars.com/kata/554b4ac871d6813a03000035
+
+
+def max_and_min(digits):
+    num_list = [int(num) for num in str(digits).split()]
```

Comment:
Does `digits` not always come in as a string?  If so how about `digits.split()`

Reply:
weird! so I tried digits.split() and in just about every way it APPEARS to do the same thing then when it gets to the first eval it checks if 10 > 10 and it obviously is false so it jumps to the next if and it checks if 10 < 9 and it evaluates that as True and then updates eval_nums[1] to what it thinks is the lesser number. 

After banging my head this is what I got in ipython:
```
In [1]: a1 = '10'

In [2]: aa1 = 10

In [3]: b1 = '9'

In [4]: bb1 = 9

In [5]: bb1 < aa1
Out[5]: True

In [6]: b1 < a1
Out[6]: False
```
So I would have to int() everything one by one?

Reply:
I tried a few things like `num_list = int(digits.split())` but in order to make that work I have to iterate over it.

and it did not like `for int(num) in num_list` nor did it like `for num in int(num_list)`

Reply:
oh my god I just realized what you meant HAHAHAHAHA. I got it now!!! I thought you literally meant replace the whole thing with digits.split(). 

But I will have you know, that I actually found that on my own after cleaning everything up and then after I found it, it hit me....OOOOHHHH that is what you meant!!! 

`    num_list = [int(num) for num in digits.split()]`

still got my head scratching for just trying to force `num_list = digits.split()` to work